src/components/organisms/ContactList.tsx

```diff
@@ -17,8 +17,9 @@ const ContactList: FC<ContactListProps> = ({ contacts, ...rest }) => {
       {!!contacts &&
         contacts.map((contact: Contact) => {
           const { rnsName, publicKey, chat } = contact
-          const nChats = !!chat && chat.length
-          const lastChatContent = !!nChats && chat[nChats - 1].content
+          const realChat = chat.filter((msg: Message) => !msg.isSync)
```

Comment:
This is a second place where you are doing filter. This can heavily impact performance.